src/store/ViewStore.ts

```diff
@@ -159,10 +160,12 @@ export default class ViewStore {
         this.mainStore.state.setChartIsReady(false);
         const sortedItems = this.sortedItems[idx].layout;
         const stx = this.stx;
-        const granularity = getIntervalInSeconds(sortedItems) as TGranularity;
+        const { interval, periodicity, timeUnit } = sortedItems;
+        const period = timeUnit ? interval : periodicity;
+        const granularity = calculateGranularity(Number(period), (timeUnit as string) || (interval as string));
```

Comment:
One of the reasons why the issue was happening was `getIntervalInSeconds()` returning a wrong `granularity` value.